From 2efd5af49d71d5ce1f0cd287bdba5905cd8a2aa6 Mon Sep 17 00:00:00 2001
From: Kevin Cernekee <cernekee@chromium.org>
Date: Wed, 11 Jan 2017 13:53:36 -0800
Subject: [PATCH] CHROMIUM: Revert "xfrm: Do not parse 32bits compiled xfrm
 netlink msg on 64bits host"

This reverts commit 74005991b78a ("xfrm: Do not parse 32bits compiled
xfrm netlink msg on 64bits host").  Our local source tree changes the
user/kernel ABI to work around the incompatibility.

BUG=chrome-os-partner:57324
TEST=run l2tp autotests on kevin, samus

Signed-off-by: Kevin Cernekee <cernekee@chromium.org>
Reviewed-on: https://chromium-review.googlesource.com/427320
Reviewed-by: Brian Norris <briannorris@chromium.org>

BUG=b:114323255
BUG=b:112641692
TEST=- Compiled, flashed grunt, booted.
     - Verified that netd's XfrmController starts correctly (logcat).
     - Verified that android.net.cts.IpSecBaseTest and
     android.net.cts.IpSecManagerTest pass.
Change-Id: I35b0c0b711b498dfe7ea764d491f2f1cf92eb304
(cherry picked from commit bf530aed496b4126a218146ae7b945fa6bd86a48)
Signed-off-by: Hugo Benichi <hugobenichi@chromium.org>
Reviewed-on: https://chromium-review.googlesource.com/1215343
Commit-Ready: Hugo Benichi <hugobenichi@google.com>
Tested-by: Hugo Benichi <hugobenichi@google.com>
Reviewed-by: Dmitry Torokhov <dtor@chromium.org>
---
 net/xfrm/xfrm_user.c | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/net/xfrm/xfrm_user.c b/net/xfrm/xfrm_user.c
index 1484bc99a537..1bb699272206 100644
--- a/net/xfrm/xfrm_user.c
+++ b/net/xfrm/xfrm_user.c
@@ -2633,11 +2633,6 @@ static int xfrm_user_rcv_msg(struct sk_buff *skb, struct nlmsghdr *nlh,
 	const struct xfrm_link *link;
 	int type, err;
 
-#ifdef CONFIG_COMPAT
-	if (in_compat_syscall())
-		return -EOPNOTSUPP;
-#endif
-
 	type = nlh->nlmsg_type;
 	if (type > XFRM_MSG_MAX)
 		return -EINVAL;
-- 
2.23.0.rc1.153.gdeed80330f-goog

